ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -118,6 +120,39 @@ void* breaker(void *d) {
   return NULL;
 }
 
+EvalParams* callback_eval_params = NULL;
+
+void mem_alloc_callback(ObjectSpace space, AllocationAction action, int size) {
+    if (NULL == callback_eval_params)
+    {
+        return; // this is probably an error that has to do with multi-threaded usage. Error somehow -seanmakesgames
+    }
+
+    if (action == kAllocationActionAllocate)
+    {
+        callback_eval_params->current_mem += size;
+        
+        printf("\nALLOC %d: %zu", size, callback_eval_params->current_mem);
+    }
+    else if (action == kAllocationActionFree)
+    {
+        callback_eval_params->current_mem -= size;
```

Comment:
Doesn't sound like the worst behavior, but I'll probably put an if in there. Reduces performance though. ;)